api/src/pcapi/core/mails/transactional/users/reset_password.py

```diff
@@ -1,31 +1,39 @@
 from pcapi.core import mails
+from pcapi.core import token as token_utils
 from pcapi.core.mails import models
 from pcapi.core.mails.transactional.sendinblue_template_ids import TransactionalEmail
 import pcapi.core.users.models as users_models
 from pcapi.utils.urls import generate_firebase_dynamic_link
 
 
-def send_reset_password_email_to_user(user: users_models.User, token: users_models.Token) -> bool:
+def send_reset_password_email_to_user(user: users_models.User, token: token_utils.Token) -> bool:
+    assert (
+        user.id == token.user_id
+    )  # we have two different sources of truth for the user, so we check that they match to prevent miss use of the function
     data = get_reset_password_email_data(user, token, TransactionalEmail.NEW_PASSWORD_REQUEST.value)
     return mails.send(recipients=[user.email], data=data)
 
 
-def send_email_already_exists_email(user: users_models.User, token: users_models.Token) -> bool:
+def send_email_already_exists_email(user: users_models.User, token: token_utils.Token) -> bool:
+    assert (
+        user.id == token.user_id
+    )  # we have two different sources of truth for the user, so we check that they match to prevent miss use of the function
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏Même suggestion ici